src/main.js

```diff
@@ -1,28 +1,68 @@
+const fs = require('fs')
+const xml2js = require('xml2js')
 const core = require('@actions/core')
-const github = require('@actions/github')
 
 /**
  * The main function for the action.
  * @returns {Promise<void>} Resolves when the action is complete.
  */
 async function run() {
-  try {
-    // The `who-to-greet` input is defined in action metadata file
-    const whoToGreet = core.getInput('who-to-greet', { required: true })
-    core.info(`Hello, ${whoToGreet}!`)
-
-    // Get the current time and set as an output
-    const time = new Date().toTimeString()
-    core.setOutput('time', time)
-
-    // Output the payload for debugging
-    core.info(
-      `The event payload: ${JSON.stringify(github.context.payload, null, 2)}`
-    )
-  } catch (error) {
-    // Fail the workflow step if an error occurs
-    core.setFailed(error.message)
-  }
+  // Read inputs of the action
+  const filePath = core.getInput('file-path', { required: true })
+  const displayOnlySurvived =
+    core.getInput('display-only-survived', { required: false }) === 'true'
+
+  const parser = new xml2js.Parser()
+  fs.readFile(filePath, (err, data) => {
+    if (err) {
+      core.setFailed(`Error reading XML file: ${err.message}`)
+      return
+    }
+
+    parser.parseString(data, (parseError, result) => {
+      if (parseError) {
+        core.setFailed(`Error parsing XML: ${parseError.message}`)
+        return
+      }
+
+      const mutationsResult = result.mutations.mutation
+      const fileGroups = mutationsResult.reduce((acc, mutation) => {
+        const file = mutation.sourceFile[0]
+        if (!acc[file]) acc[file] = []
+        const method = mutation.mutatedMethod[0]
+        const lineNumber = mutation.lineNumber[0]
+        const detected = mutation.$.detected === 'true' ? '✅' : '❌'
+        const status = mutation.$.status === 'KILLED' ? '💀' : '🚶'
+        const description = mutation.description[0]
+
+        if (!displayOnlySurvived || status === '🚶') {
+          acc[file].push(
+            `${detected} \`${method}\` (Line ${lineNumber}) - ${description} ${status}`
+          )
+        }
+        return acc
+      }, {})
+
+      let reportContent = `# Mutation Test Summary
+        
+## Overview
+
+This report provides an overview of mutation testing results, grouped by file. Each entry details a mutation attempt, its detection status, and specific mutation description.
+
+`
+
+      for (const [file, mutations] of Object.entries(fileGroups)) {
+        reportContent += `### Mutations in ${file}\n`
+        for (const mutation of mutations) {
+          reportContent += `- ${mutation}\n`
+        }
+        reportContent += '\n'
+      }
+
+      // Write directly to the GitHub Step Summary using @actions/core
+      core.summary.addRaw(reportContent).write()
+    })
+  })
```

Comment:
Formatting is a bit off, so you probably want to do  `npx prettier --write`

Reply:
I assume that GitHub did it to be agnostic to the linter usage, but it took me a while to understand that actually needed to run

```shell
$ npm run format:write
```